project_task_analytic_propagation/models/project_task.py

```diff
@@ -1,20 +1,41 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/LGPL-3.0)
 from odoo import api, fields, models
 
 
 class ProjectTask(models.Model):
-    _inherit = "project.task"
+    _name = "project.task"
+    _inherit = ["project.task", "analytic.plan.fields.mixin"]
 
-    analytic_account_id = fields.Many2one(
+    account_id = fields.Many2one(
         "account.analytic.account",
-        ondelete="set null",
-        compute="_compute_analytic_account_id",
-        store=True,
+        ondelete="restrict",
+        index=True,
+        check_company=True,
+        compute="_compute_task_analytic_accounts",
         readonly=False,
-        domain="[('company_id', '=?', company_id)]",
-        help="Analytic account to which this task and its timesheets are linked.",
+        store=True,
     )
```

Comment:
Here it also looks to me like you don't really need to repeat everything that's upstream, right?

Reply:
No, there's no upstream field there